src/auth.py

```diff
@@ -29,7 +30,8 @@ class Acl:
 class KafkaAuth:
     """Object for updating Kafka users and ACLs."""
 
-    def __init__(self, opts: List[str], zookeeper: str, container: Container):
+    def __init__(self, charm: CharmBase, opts: List[str], zookeeper: str, container: Container):
```

Comment:
Just for my sake (type-checker, helps only in this case), can we remove the hint for CharmBase?